### ISSUE TITLE
Animation Spam Log

### ISSUE THREAD

**Issue by Krimatoria** [bernie-g/geckolib#551]
Minecraft 1.20.1
Forge 47.2.6

No Animation
![2024-03-03_03 49 04](https://github.com/bernie-g/geckolib/assets/106809386/63c063eb-a5c3-44b0-a637-870f9b63c05f)

I'm a little confused, my log is littered with animation errors. I think there are two mistakes. One is related to animation. I attached a link to the other one, since my log is all in GeckoLib / Azurelib error.
https://gist.github.com/Krimatoria/8031c27bfe9281f097499676373cf00e
I once reported a problem with the animation of the SculkHorde for. I just don't know if it's the same here or new. Because at the beginning there is a mistake about GeckoLib. [Sculkhorde ](https://github.com/TeamPeril/Sculk-Horde)- they haven't said anything about this problem yet.
https://github.com/AzureDoom/AzureLib/issues/43

- sculkhorde-1.20.1-0.8.9.jar
- azurelib-neo-1.20.1-2.0.20.jar
- geckolib-forge-1.20.1-4.4.2.jar
- GeckoLibOculusCompat-Forge-1.0.1.jar
- oculus-mc1.20.1-1.6.15a.jar

I just have a big loss of TPS when I look at mobs and I thought maybe it was related to animation, I went into the log file and saw it.

> [21:23:40] [Worker-ResourceReload-4/ERROR] [so.be.ge.GeckoLib/]: Unable to parse animation: idle
> [21:23:40] [Worker-ResourceReload-0/ERROR] [so.be.ge.GeckoLib/]: Unable to parse animation: idle
> software.bernie.geckolib.core.molang.MolangException: Couldn't parse '-' expression!
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseOneLine(MolangParser.java:245)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseExpression(MolangParser.java:197)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseJson(MolangParser.java:179)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.buildKeyframeStack(BakedAnimationsAdapter.java:159)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.bakeBoneAnimations(BakedAnimationsAdapter.java:72)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.bakeAnimation(BakedAnimationsAdapter.java:53)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.deserialize(BakedAnimationsAdapter.java:39)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.deserialize(BakedAnimationsAdapter.java:31)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:76)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1214)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1319)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1261)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.FileLoader.loadAnimationsFile(FileLoader.java:28)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.cache.GeckoLibCache.lambda$loadAnimations$1(GeckoLibCache.java:90)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.cache.GeckoLibCache.lambda$loadResources$5(GeckoLibCache.java:112)
> 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
> 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
> software.bernie.geckolib.core.molang.MolangException: Couldn't parse '-' expression!
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseOneLine(MolangParser.java:245)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseExpression(MolangParser.java:197)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseJson(MolangParser.java:179)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.buildKeyframeStack(BakedAnimationsAdapter.java:159)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.bakeBoneAnimations(BakedAnimationsAdapter.java:72)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.bakeAnimation(BakedAnimationsAdapter.java:53)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.deserialize(BakedAnimationsAdapter.java:39)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.deserialize(BakedAnimationsAdapter.java:31)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:76)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1214)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1319)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1261)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.FileLoader.loadAnimationsFile(FileLoader.java:28)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.cache.GeckoLibCache.lambda$loadAnimations$1(GeckoLibCache.java:90)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.cache.GeckoLibCache.lambda$loadResources$5(GeckoLibCache.java:112)
> 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
> 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
> java.lang.Exception: Given object couldn't be converted to value! -
> 	at TRANSFORMER/mclib@20/com.eliotlash.mclib.math.MathBuilder.valueFromObject(MathBuilder.java:488)
> 	at TRANSFORMER/mclib@20/com.eliotlash.mclib.math.MathBuilder.parseSymbols(MathBuilder.java:257)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseSymbolsMolang(MolangParser.java:254)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseOneLine(MolangParser.java:242)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseExpression(MolangParser.java:197)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseJson(MolangParser.java:179)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.buildKeyframeStack(BakedAnimationsAdapter.java:159)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.bakeBoneAnimations(BakedAnimationsAdapter.java:72)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.bakeAnimation(BakedAnimationsAdapter.java:53)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.deserialize(BakedAnimationsAdapter.java:39)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.deserialize(BakedAnimationsAdapter.java:31)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:76)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1214)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1319)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1261)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.FileLoader.loadAnimationsFile(FileLoader.java:28)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.cache.GeckoLibCache.lambda$loadAnimations$1(GeckoLibCache.java:90)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.cache.GeckoLibCache.lambda$loadResources$5(GeckoLibCache.java:112)
> 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
> 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
> software.bernie.geckolib.core.molang.MolangException: Couldn't parse '-' expression!
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseOneLine(MolangParser.java:245)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseExpression(MolangParser.java:197)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.core.molang.MolangParser.parseJson(MolangParser.java:179)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.buildKeyframeStack(BakedAnimationsAdapter.java:159)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.bakeBoneAnimations(BakedAnimationsAdapter.java:72)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.bakeAnimation(BakedAnimationsAdapter.java:53)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.deserialize(BakedAnimationsAdapter.java:39)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.json.typeadapter.BakedAnimationsAdapter.deserialize(BakedAnimationsAdapter.java:31)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:76)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1214)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1319)
> 	at MC-BOOTSTRAP/com.google.gson@2.10/com.google.gson.Gson.fromJson(Gson.java:1261)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.loading.FileLoader.loadAnimationsFile(FileLoader.java:28)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.cache.GeckoLibCache.lambda$loadAnimations$1(GeckoLibCache.java:90)
> 	at TRANSFORMER/geckolib@4.4.2/software.bernie.geckolib.cache.GeckoLibCache.lambda$loadResources$5(GeckoLibCache.java:112)
> 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
> 	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.exec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinTask.doExec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool.scan(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinPool.runWorker(Unknown Source)
> 	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(Unknown Source)
> [21:23:40] [Worker-ResourceReload-0/ERROR] [so.be.ge.GeckoLib/]: Unable to parse animation: idle
>

**Comment by Tslat** [comment]
Yes like I said you have an invalid animation json

**Comment by Krimatoria** [comment]
> Да, как я уже сказал, у вас неверный JSON анимации.

I don't know if you're interested or not) But I found one of the problematic animations. The problem was in the most prominent place in the screenshot above. I just removed born_in_chaos_[Forge]1.20.1_1.2.jar and part of the messages of Unable to parse animation: idle disappeared from the log and only one remained. I.e., each Unable to parse animation: idle is associated with some kind of mobs in fashion.

I will inform them that something is wrong with their animation. But there are still mods left to find that affect animation.
